File: 1160-find-words-that-can-be-formed-by-characters/1160-find-words-that-can-be-formed-by-characters.cpp
class Solution {
public:
    bool isGood(string str, string chars) {
        map<char, int> m;
        for(int i = 0; i < chars.length(); i++) 
            m[chars[i]]++;
        
        for(int i = 0; i < str.length(); i++) {
            if( m[str[i]] == 0)
                return false;
            else
                m[str[i]]--;
        }
        return true;
    }
    int countCharacters(vector<string>& words, string chars) {
        int sum = 0;
        for(int i = 0; i < words.size(); i++){
            if(isGood(words[i], chars))
                sum += words[i].length();
        }
        return sum;
    }
};

File: 1160-find-words-that-can-be-formed-by-characters/1160-find-words-that-can-be-formed-by-characters.py
class Solution(object):
    def countCharacters(self, words, chars):
        """
        :type words: List[str]
        :type chars: str
        :rtype: int
        """
        sum = 0
        for s in words:
            flag = True;
            for c in s:
                if chars.count(c) < s.count(c):
                    flag = False
            if flag:
                sum += len(s)
        
        return sum